packages/react/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,580 @@
+                );
+                this.resizeDetector.observe(this.rootElement.current);
+            }
+        }
+
+        this.viewportSpan = this.getViewportSpan();
+        this.updateLayout();
+
+        // if we need to have a particular child scrolled into view on mount do so
+        if (
+            !isNil(this.props.initiallyVisibleItemIndex) &&
+            this.itemPositions.length > this.props.initiallyVisibleItemIndex
+        ) {
+            const newScrollPosition: number = this.getScrollIntoViewPosition(
+                this.props.initiallyVisibleItemIndex
+            );
+            this.setScrollPosition(newScrollPosition);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(prevProps: StackPanelProps): void {
+        if (prevProps.orientation !== this.props.orientation) {
+            this.viewportSpan = this.getViewportSpan();
+        }
+
+        if (
+            prevProps.itemSpan !== this.props.itemSpan ||
+            prevProps.orientation !== this.props.orientation ||
+            prevProps.children !== this.props.children
+        ) {
+            this.updateItemPositions();
+            this.updateLayout();
+        }
+
+        // if the initiallyVisibleItemIndex prop is changed to a non-null value
+        // after initial mount we scroll to that position
+        if (
+            prevProps.initiallyVisibleItemIndex !==
+                this.props.initiallyVisibleItemIndex &&
+            !isNil(this.props.initiallyVisibleItemIndex) &&
+            this.itemPositions.length > this.props.initiallyVisibleItemIndex
+        ) {
+            const newScrollPosition: number = this.getScrollIntoViewPosition(
+                this.props.initiallyVisibleItemIndex
+            );
+            this.scrollContent(newScrollPosition);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentWillUnmount(): void {
+        if (!isNil(this.rootElement.current)) {
+            if (((window as unknown) as WindowWithResizeObserver).ResizeObserver) {
+                this.resizeDetector.disconnect();
+                this.resizeDetector = null;
+            }
+        }
+        clearTimeout(this.scrollLayoutUpdateTimer);
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        const {
+            stackPanel,
+            stackPanel__scrollable,
+        }: StackPanelClassNameContract = this.props.managedClasses;
+
+        return super.generateClassNames(
+            classNames(stackPanel, [stackPanel__scrollable, this.state.isScrollable])
+        );
+    }
+
+    /**
+     * Render the items
+     */
+    private renderItems(): React.ReactFragment[] {
+        return React.Children.map(this.props.children, this.renderItem);
+    }
+
+    /**
+     * Test if a node is cloneable
+     */
+    private isClonableElement(node: React.ReactNode): node is React.ReactElement<any> {
```

Comment:
Do you really need to alias the React function? `React.isValidElement` is a type guard

Reply:
nah, removed